test/e2e_node/topology_manager_test.go

```diff
@@ -0,0 +1,388 @@
+	err = f.PodClient().MatchContainerOutput(pod.Name, pod.Spec.Containers[0].Name, expAllowedCPUsListRegex)
+	framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+		pod.Spec.Containers[0].Name, pod.Name)
+
+	err = f.PodClient().MatchContainerOutput(pod.Name, pod.Spec.Containers[0].Name, expAllowedCPUsListRegex)
+	framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+		pod.Spec.Containers[1].Name, pod.Name)
+
+	ginkgo.By("by deleting the pods and waiting for container removal")
+	deletePods(f, []string{pod.Name})
+	waitForContainerRemoval(pod.Spec.Containers[0].Name, pod.Name, pod.Namespace)
+	waitForContainerRemoval(pod.Spec.Containers[1].Name, pod.Name, pod.Namespace)
+
+	ginkgo.By("running multiple Gu pods")
+	ctnAttrs = []tmCtnAttribute{
+		{
+			ctnName:    "gu-container1",
+			cpuRequest: "1000m",
+			cpuLimit:   "1000m",
+		},
+	}
+	pod1 = makeTopologyManagerPod("gu-pod1", ctnAttrs)
+	pod1 = f.PodClient().CreateSync(pod1)
+
+	ctnAttrs = []tmCtnAttribute{
+		{
+			ctnName:    "gu-container2",
+			cpuRequest: "1000m",
+			cpuLimit:   "1000m",
+		},
+	}
+	pod2 = makeTopologyManagerPod("gu-pod2", ctnAttrs)
+	pod2 = f.PodClient().CreateSync(pod2)
+
+	ginkgo.By("checking if the expected cpuset was assigned")
+	cpu1, cpu2 = 1, 2
+	if isHTEnabled() {
+		cpuList = cpuset.MustParse(getCPUSiblingList(0)).ToSlice()
+		if cpuList[1] != 1 {
+			cpu1, cpu2 = cpuList[1], 1
+		}
+	}
+
+	expAllowedCPUsListRegex = fmt.Sprintf("^%d\n$", cpu1)
+	err = f.PodClient().MatchContainerOutput(pod1.Name, pod1.Spec.Containers[0].Name, expAllowedCPUsListRegex)
+	framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+		pod1.Spec.Containers[0].Name, pod1.Name)
+
+	expAllowedCPUsListRegex = fmt.Sprintf("^%d\n$", cpu2)
+	err = f.PodClient().MatchContainerOutput(pod2.Name, pod2.Spec.Containers[0].Name, expAllowedCPUsListRegex)
+	framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+		pod2.Spec.Containers[0].Name, pod2.Name)
+
+	ginkgo.By("by deleting the pods and waiting for container removal")
+	deletePods(f, []string{pod1.Name, pod2.Name})
+	waitForContainerRemoval(pod1.Spec.Containers[0].Name, pod1.Name, pod1.Namespace)
+	waitForContainerRemoval(pod2.Spec.Containers[0].Name, pod2.Name, pod2.Namespace)
+}
+
+func runTopologyManagerTests(f *framework.Framework) {
+	var oldCfg *kubeletconfig.KubeletConfiguration
+
+	ginkgo.It("run Topology Manager test suite", func() {
+
+		var policies = []string{topologymanager.PolicySingleNumaNode, topologymanager.PolicyRestricted,
+			topologymanager.PolicyBestEffort, topologymanager.PolicyNone}
+
+		for _, policy := range policies {
+			// Configure Topology Manager
+			ginkgo.By("by configuring Topology Manager policy to xxx")
+			framework.Logf("Configuring topology Manager policy to %s", policy)
+			configureTopologyManagerInKubelet(f, policy)
+			// Run the tests
+			runTopologyManagerSuiteTests(f)
```

Comment:
I see now what this is doing, so we basically configure topology manager policies, and then run the same basic tests as we cover in cpu manager test suite with a corresponding topology manager.  this makes sense as a first starting point, but i assume we will have a follow-on PR to actually validate topology policy applied if/when we add a device?

Reply:
Yes, here is the PR to add checking for alignment of CPU and SR-IOV devices:
https://github.com/kubernetes/kubernetes/pull/87645

Reply:
Hi, I updated my PR with more positive test and negative test. I'm more than open about suggestions about adding more positive and negative tests, please review and suggest!